### ISSUE TITLE
Single-genome prediction produces compressed probability scores

### ISSUE THREAD

**Issue by duriah** [LorMeBioAI/DCiPatho#5]
Hi
I am trying to use the tool on a single complete genome, but with no success. Any help is appreciated

## Environment
- DCiPatho model: `DCiPatho_best_k3-7_model.pt` (Zenodo 7681330)
- Python 3.7, torch 1.10.0, numpy 1.20.3
- k-mers: 3,4,5,6,7

## Problem
When predicting single genomes, DCiPatho produces compressed probability scores (0.50-0.73 range) and misclassifies known non-pathogens.

**Test results:**
- Known non-pathogen (*Lactobacillus acidophilus*): **0.50** (classified as pathogenic)
- Various test genomes: 0.50-0.73 range

**Expected:** Wide distribution 0.0-1.0, with non-pathogens scoring <0.3

### Approach 1: Using raw k-mer frequencies (no standardization)
```python
predictions = model(torch.tensor(features).float())
```
**Result:** Constant output **0.56** for all genomes (likely model baseline)

### Approach 2: Standardizing based on single genome (as in original code)
```python
scaler = StandardScaler().fit(features)  # Fit on single genome
X_scaled = scaler.transform(features)
predictions = model(torch.tensor(X_scaled).float())
```
**Result:** Constant output **0.56** for all genomes
- Produces all-zeros vector (std=0 for single sample)
- Model receives no information

### Approach 3: Using stored training scaler parameters
```python
# Extracted from toy_dataset_for_DCiPatho.zip (Zenodo 7681330)
# Combined toy_patho_freq.csv + toy_nonpatho_freq.csv (1,506 genomes)
# Computed mean and std across all training samples for each feature
scaler_mean = np.load('scaler_mean.npy')  # Shape: (10952,)
scaler_std = np.load('scaler_std.npy')    # Shape: (10952,)

X_scaled = (features - scaler_mean) / scaler_std
predictions = model(torch.tensor(X_scaled).float())
```
**Result:** Compressed scores (0.50-0.73)
- Known non-pathogen (*Lactobacillus*): 0.50
- This seems like the most reasonable approach, but scores remain compressed

## Questions

1. **Is single-genome prediction supported?** Or does DCiPatho require batch prediction?

2. **What is the correct preprocessing for single genomes?** How should we standardize when predicting one genome at a time?

3. **How were the model weights trained?** What preprocessing was applied to the training data? 

4. **Are the stored scaler parameters available?** If the model expects specific mean/std values, where can we obtain them?

## Request
Could you please provide guidance on the correct workflow for predicting a single genome? Understanding the proper preprocessing approach is critical for our genomic pipeline where genomes are processed individually.

Thank you for this excellent tool and for your assistance!

**Comment by duriah** [comment]
Given the lack of response, I validated DCiPatho on 300 genomes against MBPD classifications using approach 3 (see above). Despite compressed raw scores (~0.50-0.73), the model discriminates well:

    Pathogens: mean 0.70, Non-pathogens: mean 0.52
    Cohen's d = 3.2, p < 2.2e-16

Thus I decided to simply normalize the values based on this range.

<img width="2937" height="1654" alt="Image" src="https://github.com/user-attachments/assets/084f345d-e09d-4e17-91c4-6a4f352169f8" />
